src/Framework/WordPressLibraries/WPBackgroundProcess.php

```diff
@@ -0,0 +1,533 @@
+            wp_die();
+        }
+
+        check_ajax_referer($this->identifier, 'nonce');
+
+        $this->handle();
+
+        wp_die();
+    }
+
+    /**
+     * Is queue empty
+     *
+     * @return bool
+     */
+    protected function is_queue_empty()
+    {
+        global $wpdb;
+
+        $table = $wpdb->options;
+        $column = 'option_name';
+
+        if (is_multisite()) {
+            $table = $wpdb->sitemeta;
+            $column = 'meta_key';
+        }
+
+        $key = $this->identifier . '_batch_%';
+
+        $count = $wpdb->get_var(
+            $wpdb->prepare(
+                "
+			SELECT COUNT(*)
+			FROM {$table}
+			WHERE {$column} LIKE %s
+		",
+                $key
+            )
+        );
+
+        return ($count > 0) ? false : true;
+    }
+
+    /**
+     * Is process running
+     *
+     * Check whether the current process is already running
+     * in a background process.
+     */
+    public function is_process_running()
+    {
+        if (get_site_transient($this->identifier . '_process_lock')) {
+            // Process already running.
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Lock process
+     *
+     * Lock the process so that multiple instances can't run simultaneously.
+     * Override if applicable, but the duration should be greater than that
+     * defined in the time_exceeded() method.
+     */
+    protected function lock_process()
+    {
+        $this->start_time = time(); // Set start time of current process.
+
+        $lock_duration = (property_exists($this, 'queue_lock_time')) ? $this->queue_lock_time : 60; // 1 minute
+        $lock_duration = apply_filters($this->identifier . '_queue_lock_time', $lock_duration);
+
+        set_site_transient($this->identifier . '_process_lock', microtime(), $lock_duration);
+    }
+
+    /**
+     * Unlock process
+     *
+     * Unlock the process so that other instances can spawn.
+     *
+     * @return $this
+     */
+    protected function unlock_process()
+    {
+        delete_site_transient($this->identifier . '_process_lock');
+
+        return $this;
+    }
+
+    /**
+     * Get batch
+     *
+     * @return stdClass Return the first batch from the queue
+     */
+    protected function get_batch()
+    {
+        global $wpdb;
+
+        $table = $wpdb->options;
+        $column = 'option_name';
+        $key_column = 'option_id';
+        $value_column = 'option_value';
+
+        if (is_multisite()) {
+            $table = $wpdb->sitemeta;
+            $column = 'meta_key';
+            $key_column = 'meta_id';
+            $value_column = 'meta_value';
+        }
+
+        $key = $this->identifier . '_batch_%';
+
+        $query = $wpdb->get_row(
+            $wpdb->prepare(
+                "
+			SELECT *
+			FROM {$table}
+			WHERE {$column} LIKE %s
+			ORDER BY {$key_column} ASC
+			LIMIT 1
+		",
+                $key
+            )
+        );
+
+        $batch = new stdClass();
```

Comment:
Static analysis caught this - since the class is now within a namespace I think this needs to be:
```suggestion
        $batch = new \stdClass();
```

Reply:
I see thanks! Resolved: f4c900c1a557a77ed5a3497d3e3edae12f02b608